qiskit/transpiler/passes/optimization/collect_cliffords.py

```diff
@@ -69,21 +70,9 @@ def __init__(
         )
 
 
-clifford_gate_names = [
-    "x",
-    "y",
-    "z",
-    "h",
-    "s",
-    "sdg",
-    "cx",
-    "cy",
-    "cz",
-    "swap",
-    "clifford",
-    "linear_function",
-    "pauli",
-]
+clifford_gate_names = (
+    list(_BASIS_1Q.keys()) + list(_BASIS_2Q.keys()) + ["clifford", "linear_function", "pauli"]
```

Comment:
should we also add "permutation" here?

Reply:
I think it's a good idea. I have checked and we can indeed construct a Clifford from a circuit containing `PermutationGate` objects, so it makes sense for the collection pass to handle these as well